packages/core/useScrollParent/index.ts

```diff
@@ -0,0 +1,81 @@
+import type { ComputedRef, MaybeRefOrGetter } from 'vue'
+import type { ConfigurableDocument } from '../_configurable'
+import { computed, toValue } from 'vue'
+import { defaultDocument } from '../_configurable'
+
+export interface UseScrollParentOptions extends ConfigurableDocument {
+  /**
+   * The number of parent levels to lookup
+   *
+   * @default 10
+   */
+  lookupLevel?: number
+  /**
+   * On error callback
+   *
+   * Default log error to `console.error`
+   */
+  onError?: (error: unknown) => void
+}
+
+export type ScrollParent = Element | null | undefined
+
+/**
+ * Reactive nearest scrollable parent of the element
+ *
+ * @see https://vueuse.org/useScrollParent
+ * @param element
+ * @param options
+ */
+export function useScrollParent(
+  element: MaybeRefOrGetter<ScrollParent>,
+  options: UseScrollParentOptions = {},
+): ComputedRef<ScrollParent> {
+  const {
+    document: root = defaultDocument,
+    lookupLevel = 10,
+    onError = (e) => { console.error(e) },
+  } = options
+  const overflowScrollReg = /scroll|auto|overlay/i
```

Comment:
```suggestion
  const baseRegex = 'scroll|overlay|auto'
  const overflowScrollReg = includeHidden
    ? new RegExp(`${baseRegex}|hidden`, 'i')
     : new RegExp(baseRegex, 'i')
```

With this, we can include visually hidden elements, which I think it's a good addition. ``includeHidden`` must be an option.

Reply:
@yue1123 You forgot about this one or you don't agree with me in the need for this?

Reply:
@yue1123 Yes, exactly. I'm not 100% sure of the use case for this as well, but the virtual scroll library I'm using [takes this into account](https://github.com/rocwang/vue-virtual-scroll-grid/blob/main/src/utilites.ts#L98), [jQuery](https://github.com/fis-components/jquery-ui/blob/90fe5b0943d49c2369ad43fe9d6dba26f63f410f/core.js#L42) does as well

Reply:
@ferferga OK, I see. You’ve thought it through thoroughly — this situation can definitely happen, even if it’s a bit of an edge case. I’ll add an option to control it later, as you suggested.

Reply:
@ferferga If there’s any issue with the code in the PR, please point it out and I’ll make adjustments promptly, thank you.